tests/autologging/test_autologging_utils.py

```diff
@@ -642,3 +644,134 @@ def test_autologging_event_logger_default_impl_warns_for_log_autolog_called_with
             call_args=("a"),
             call_kwargs={"b": "c"},
         )
+
+
+def test_is_autologging_supported():
+    assert _check_version_in_range("1.0.2", "1.0.1", "1.0.3")
+    assert _check_version_in_range("1.0.1", "1.0.1", "1.0.3")
+    assert _check_version_in_range("1.0.3", "1.0.1", "1.0.3")
+    assert not _check_version_in_range("1.0.0", "1.0.1", "1.0.3")
+    assert not _check_version_in_range("1.0.4", "1.0.1", "1.0.3")
+    assert not _check_version_in_range("0.99.99", "1.0.1", "1.0.3")
+    assert not _check_version_in_range("1.1.0", "1.0.1", "1.0.3")
+    assert _check_version_in_range("1.0.3", "1.0.1", "1.0.3.post1")
+
+    def gen_get_module_version_fn(module_name_to_version_map):
+        def get_model_version_fn(module_name):
+            if module_name in module_name_to_version_map:
+                return module_name_to_version_map[module_name]
+            raise ValueError("Test error: unknown module name: " + module_name)
+
+        return get_model_version_fn
+
+    assert _is_autologging_integration_supported(
+        "fastai", gen_get_module_version_fn({"fastai": "1.0.60"})
+    )
```

Comment:
Can we use parametrization to make it eaisier to debug test failures + make the code DRY?

https://docs.pytest.org/en/stable/parametrize.html#parametrizing-fixtures-and-test-functions